Xamarin.Essentials/Connectivity/Connectivity.ios.cs

```diff
@@ -1,36 +1,40 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Threading.Tasks;
+﻿using System.Collections.Generic;
 
 namespace Xamarin.Essentials
 {
     public static partial class Connectivity
     {
-        static void StartListeners() =>
-            Reachability.ReachabilityChanged += ReachabilityChanged;
+        static ReachabilityListener listener;
 
-        static async void ReachabilityChanged(object sender, EventArgs e)
+        static void StartListeners()
         {
-            await Task.Delay(100);
-            OnConnectivityChanged();
+            listener = new ReachabilityListener();
+            listener.ReachabilityChanged += OnConnectivityChanged;
```

Comment:
No longer a static implementation, the listeners are created and disposed when events are attached or removed.